homeassistant/components/sensoterra/sensor.py

```diff
@@ -0,0 +1,146 @@
+"""Sensoterra devices."""
+
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONFIGURATION_URL, DOMAIN
+from .coordinator import SensoterraCoordinator, SensoterraSensor
+
+SENSORS: dict[str, SensorEntityDescription] = {
+    "MOISTURE": SensorEntityDescription(
+        key="MOISTURE",
+        has_entity_name=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.MOISTURE,
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key="soil_moisture_at_cm",
+    ),
+    "SI": SensorEntityDescription(
+        key="SI",
+        has_entity_name=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        name="SI @ {depth} cm",
+    ),
+    "TEMPERATURE": SensorEntityDescription(
+        key="TEMPERATURE",
+        has_entity_name=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    "BATTERY": SensorEntityDescription(
+        key="BATTERY",
+        has_entity_name=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "RSSI": SensorEntityDescription(
+        key="RSSI",
+        has_entity_name=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "LASTSEEN": SensorEntityDescription(
+        key="LASTSEEN",
+        has_entity_name=True,
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        translation_key="last_seen",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_devices: AddEntitiesCallback
+) -> None:
+    """Set up Sensoterra sensor."""
+
+    coordinator: SensoterraCoordinator = entry.runtime_data
+
+    def _add_devices(devices: dict[str, SensoterraSensor]) -> None:
```

Comment:
```suggestion
    @callback
    def _async_add_devices(devices: dict[str, SensoterraSensor]) -> None:
```
is not blocking, so we can mark it as such